go/tools/gazelle/config/directives.go

```diff
@@ -107,10 +108,89 @@ func ApplyDirectives(c *Config, directives []Directive) *Config {
 		case "build_file_name":
 			modified.ValidBuildFileNames = strings.Split(d.Value, ",")
 			didModify = true
+		case "proto":
+			protoMode, err := ProtoModeFromString(d.Value)
+			if err != nil {
+				log.Print(err)
+				continue
+			}
+			modified.ProtoMode = protoMode
+			didModify = true
 		}
 	}
 	if !didModify {
 		return c
 	}
 	return &modified
 }
+
+// InferProtoMode sets Config.ProtoMode, based on the contents of f.  If the
+// proto mode is already set to something other than the default, or if the mode
+// is set explicitly in directives, this function does nothing. If the legacy
+// go_proto_library.bzl is loaded, or if this is the Well Known Types
+// repository, legacy mode is used.  If go_proto_library is loaded from another
```

Comment:
This part of the description implies it would not mantain new proto rules that it generated, but looking at the code below you stay in DefaultMode in that case.

Reply:
Done. This was the result of a last-minute change.